packages/core/src/diff.rs

```diff
@@ -160,6 +160,10 @@ impl<'b> VirtualDom {
                                             self.diff_attribute(left, right, mounted_id)
                                         }
                                     }
+
+                                    // consume both
+                                    left_iter.next();
```

Comment:
I don't think consuming both items is the right behavior for the greater or less branches. We should only consume the attribute that is less than the other attribute.

If we have the attributes:
old_attributes: [`aaaa: "world"`, `aaab: "world"`]
new_attributes: [`aaab: "world"`]

first iteration:
left = `aaaa: "world"`
right `aaab: "world"`
left < right, so we remove the attribute "aaaa: world"

If we move both iterators forward one item, then the remaining iterators would have the items:
old_attributes: [`aaab: "world"`]
new_attributes: []

In the second iteration, it would incorrectly try to create the attribute `aaab: "world"` instead of diffing it

Reply:
Yep, overlooked that bit.

However, the approach of consuming the smaller value assumes the elements are sorted by name, right? Also, the Miri test fails due to a memory leak from `cx.bump().alloc()` (i think?).

Any suggestions for an alternative approach to allocate the attributes for the test?

Reply:
> Yep, overlooked that bit.
> 
> However, the approach of consuming the smaller value assumes the elements are sorted by name, right? 

Yes, they should be sorted by name in the rsx macro

> Also, the Miri test fails due to a memory leak from `cx.bump().alloc()` (i think?).
> 
> Any suggestions for an alternative approach to allocate the attributes for the test?

If you allocate anything in the bump allocator, it will not be dropped. That is fine for values like numbers and simple structs, but boxes and vecs rely on the drop implementation so we need to avoid them. You can allocate individual slides with `cx.bump().allow([your slice])` instead of allocating a vec for each branch to fix the issue